resource-sharing-partners/src/test/java/no/sikt/rsp/ResourceSharingPartnerTest.java

```diff
@@ -1,11 +1,209 @@
+        assertThat(appender.getMessages(), containsString(expectedMessage));
+    }
+
+    @Test
+    public void shouldLogExceptionWhenS3BucketFileCannotBeConvertedToBaseBibliotek() throws IOException {
+        var uri = s3Driver.insertFile(randomS3Path(), INVALID_BASEBIBLIOTEK_XML_STRING);
+        var s3Event = createS3Event(uri);
+        assertThrows(RuntimeException.class, () -> resourceSharingPartnerHandler.handleRequest(s3Event, CONTEXT));
+    }
+
+    @Test
+    public void shouldBeAbleToConvertFullBasebibliotekFile() throws IOException {
+        var fullBaseBibliotekFile = IoUtils.stringFromResources(Path.of(BASEBIBLIOTEK_XML));
+        var uri = s3Driver.insertFile(randomS3Path(), fullBaseBibliotekFile);
+        var s3Event = createS3Event(uri);
+        assertDoesNotThrow(() -> resourceSharingPartnerHandler.handleRequest(s3Event, CONTEXT));
+    }
+
+    @Test
+    public void basebibliotekMissingLibNrOrLandkodeShouldNotBeConvertedToBasebibliotekBean() throws IOException {
+
+        var specifiedList = List.of(
+            new RecordSpecification(true, true, null, randomBoolean(), randomBoolean()),
+            new RecordSpecification(false, false, null, randomBoolean(), randomBoolean()),
+            new RecordSpecification(true, false, null, randomBoolean(), randomBoolean()),
+            new RecordSpecification(false, true, null, randomBoolean(), randomBoolean()));
+        var basebibliotekGenerator = new BasebibliotekGenerator(specifiedList);
+        var basebibliotek = basebibliotekGenerator.generateBaseBibliotek();
+        var basebibliotekXml = BasebibliotekGenerator.toXml(basebibliotek);
+        var uri = s3Driver.insertFile(randomS3Path(), basebibliotekXml);
+        var s3Event = createS3Event(uri);
+        var basebibliotekBeans = resourceSharingPartnerHandler.handleRequest(s3Event, CONTEXT);
+        assertThat(basebibliotekBeans, hasSize(1));
+        assertBaseBibliotekBeansHasExtractedSimppleMatadataCorrectly(basebibliotekBeans.get(0),
+                                                                     basebibliotek.getRecord().get(0));
+    }
+
+    @Test
+    public void basebibliotekBeanShouldContainNncipUri() throws IOException {
+        var expectedUri = randomUri().toString();
+        var specifiedList = List.of(
+            new RecordSpecification(true, true, expectedUri, randomBoolean(), randomBoolean()));
+
+        var basebibliotekGenerator = new BasebibliotekGenerator(specifiedList);
+        var basebibliotek = basebibliotekGenerator.generateBaseBibliotek();
+        var basebibliotekXml = BasebibliotekGenerator.toXml(basebibliotek);
+        var uri = s3Driver.insertFile(randomS3Path(), basebibliotekXml);
+        var s3Event = createS3Event(uri);
+        var basebibliotekBeans = resourceSharingPartnerHandler.handleRequest(s3Event, CONTEXT);
+        assertEquals(expectedUri, basebibliotekBeans.get(0).getNncippServer());
+    }
```

Comment:
assertThat(basebibliotekBeans.get(0).getNncippServer(), is(equalTo(expectedUri)))

Reply:
fjernet assertEquals og byttet til assertThat